records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,200 @@
+    hint_quoting: HintQuoting = hints['quoting']  # type: ignore
+    if hint_quoting == 'all':
+        mysql_field_enclosed_by = hint_quotechar
+    elif hint_quoting == 'minimal':
+        # "If the input values are not necessarily enclosed within
+        # quotation marks, use OPTIONALLY before the ENCLOSED BY option."
+        #
+        # This implies to me that parsing here is permissive -
+        # otherwise unambiguous strings without double quotes around
+        # them will be understood as a string, not rejected.
+        mysql_field_optionally_enclosed_by = hint_quotechar
+    elif hint_quoting == 'nonnumeric':
+        mysql_field_optionally_enclosed_by = hint_quotechar
+    elif hint_quoting is None:
+        pass
+    else:
+        # TODO: Have this be more like _assert_never(hint_quoting)?
+        cant_handle_hint(fail_if_cant_handle_hint, 'quoting', hints)
+        # TODO: Can we do this here?  Does it make sense?   _assert_never(hint_quoting)
+    quiet_remove(unhandled_hints, 'quoting')
+    quiet_remove(unhandled_hints, 'quotechar')
+
+    # If the field begins with the ENCLOSED BY character, instances of
+    # that character are recognized as terminating a field value only
+    # if followed by the field or line TERMINATED BY sequence. To
+    # avoid ambiguity, occurrences of the ENCLOSED BY character within
+    # a field value can be doubled and are interpreted as a single
+    # instance of the character. For example, if ENCLOSED BY '"' is
+    # specified, quotation marks are handled as shown here:
+    hint_doublequote: HintDoublequote = hints['doublequote']  # type: ignore
+    if hint_quoting is not None:
+        if hint_doublequote is True:
+            pass
+        elif hint_doublequote is False:
+            cant_handle_hint(fail_if_cant_handle_hint, 'doublequote', hints)
+        else:
+            # TODO: Have this be more like _assert_never(hint_quoting)?
+            cant_handle_hint(fail_if_cant_handle_hint, 'doublequote', hints)
+    quiet_remove(unhandled_hints, 'doublequote')
+
+    # FIELDS ESCAPED BY controls how to read or write special characters:
+    #
+    # * For input, if the FIELDS ESCAPED BY character is not empty,
+    #   occurrences of that character are stripped and the following
+    #   character is taken literally as part of a field value. Some
+    #   two-character sequences that are exceptions, where the first
+    #   character is the escape character.
+    #
+    # [...]
+    #
+    # If the FIELDS ESCAPED BY character is empty, escape-sequence
+    # interpretation does not occur.
+    hint_escape: HintEscape = hints['escape']  # type: ignore
+    if hint_escape is None:
+        mysql_field_escaped_by = None
+    elif hint_escape == '\\':
+        mysql_field_escaped_by = '\\'
+    else:
+        cant_handle_hint(fail_if_cant_handle_hint, 'escape', hints)
+        # TODO: Can we do this here?  Does it make sense?   _assert_never(hint_quoting)
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker